addons/html_builder/static/src/plugins/google_map_option/google_map_option_plugin.js

```diff
@@ -0,0 +1,331 @@
+import { GoogleMapOption } from "./google_map_option";
+
+/**
+ * A `google.maps.places.PlaceResult` object.
+ * Here listed are only the few properties used here. For a full list, see:
+ * {@link https://developers.google.com/maps/documentation/javascript/reference/places-service#PlaceResult}
+ *
+ * @typedef {Object} Place
+ * @property {string} [formatted_address]
+ * @property {Object} [geometry]
+ * @property {Object} [geometry.location]
+ * @property {function():number} geometry.location.lat
+ * @property {function():number} geometry.location.lng
+ */
+/**
+ * A string defining GPS coordinates in the form "`Latitude`,`Longitude`".
+ * @typedef {`${number},${number}`} Coordinates
+ */
+/**
+ * @typedef {{ isValid: boolean, message?: string }} ApiKeyValidation
+ */
+
+class GoogleMapOptionPlugin extends Plugin {
+    static id = "googleMapOption";
+    static dependencies = ["remove"];
+    resources = {
+        builder_options: [
+            {
+                OptionComponent: GoogleMapOption,
+                selector: ".s_google_map",
+                props: {
+                    loadGoogleMaps: this.loadGoogleMaps.bind(this),
+                    commitPlace: this.commitPlace.bind(this),
+                },
+            },
+        ],
+        builder_actions: this.getActions(),
+    };
+
+    setup() {
+        this.websiteService = this.services.website;
+        // @TODO mysterious-egg: the `google_map service` is a duplicate of the
+        // `website_map_service`, but without the dependency on public
+        // interactions. These are used only to restart the interactions once
+        // the API is loaded. We do this in the plugin instead. Once
+        // `html_builder` replaces `website`, we should be able to remove
+        // `website_map_service` since only google_map service will be used.
+        this.googleMapService = this.services.google_map;
+        this.dialog = this.services.dialog;
+        this.orm = this.services.orm;
+        this.notification = this.services.notification;
+
+        /** @type {Map<Coordinates, Place>} */
+        this.gpsMapCache = new Map();
+    }
+
+    destroy() {
+        // Without this, the google library injects elements inside the backend
+        // DOM but do not remove them once the editor is left. Notice that
+        // this is also done when the widget is destroyed for another reason
+        // than leaving the editor, but if the google API needs that container
+        // again afterwards, it will simply recreate it.
+        for (const container of document.body.querySelectorAll('.pac-container')) {
+            container.remove();
+        }
+    }
+
+    getActions() {
+        return {
+            resetMapColor: {
+                apply: ({ editingElement }) => {
+                    editingElement.dataset.mapColor = "";
+                },
+            },
+            showDescription: {
+                apply: ({ editingElement }) => {
```

Comment:
you need to define isApplied:() =>  editingElement.querySelector(".description")
clean: () => descriptionEl.remove()
apply () => append(...)